Src/Base/AMReX_TimeIntegrator.H

```diff
@@ -0,0 +1,120 @@
+#ifndef AMREX_TIME_INTEGRATOR_H
+#define AMREX_TIME_INTEGRATOR_H
+#include <AMReX_REAL.H>
+#include <AMReX_Vector.H>
+#include <AMReX_ParmParse.H>
+#include <AMReX_IntegratorBase.H>
+#include <AMReX_FEIntegrator.H>
+#include <AMReX_RKIntegrator.H>
+#include <functional>
+
+namespace amrex {
+
+namespace IntegratorTypes {
+    enum TimeIntegratorTypes {ForwardEuler = 0,
+                              ExplicitRungeKutta};
+};
+
+template<class T>
+class TimeIntegrator
+{
+private:
+    std::unique_ptr<IntegratorBase<T> > integrator_ptr;
+    std::function<void ()> post_timestep;
+
+public:
+    TimeIntegrator(T& S_data)
+    {
+        amrex::ParmParse pp("integration");
```

Comment:
It would be nice to have a constructor that does not use ParmParse.

Reply:
I agree ... will work on it